### ISSUE TITLE
Changing hillshade-illumination-direction near 0/360 triggers page refresh

### ISSUE THREAD

**Issue by stevage** [mapbox/mapbox-gl-js#5934]
**mapbox-gl-js version**: 0.43.0

### Steps to Trigger Behavior

 1. Create a hilshade layer with `hillshade-illumination-direction` of 359
 2. Call `map.setPaintProperty(layername, 'hillshade-illumination-direction', 0)`

### Expected Behavior
Almost no visible effect.

### Actual Behavior
Entire map repaints with large visible flicker.

Easily seen here: https://bl.ocks.org/stevage/raw/a4569595d8592181ab09a5a13166be8d

(The current direction is shown in the URL)

Code: https://gist.github.com/stevage/a4569595d8592181ab09a5a13166be8d

**Comment by mollymerp** [comment]
yes I see what you mean – I don't think it's triggering a page refresh though, its just transitioning from 359 -> 0 counterclockwise so the light source does almost a full rotation instead of just moving slightly clockwise to 360/0. 

A workaround right now is to set `transition.duration` to 0 https://www.mapbox.com/mapbox-gl-js/style-spec#root-transition

**Comment by stevage** [comment]
Ah, that seems so obvious now :) Thanks.
